controllers/prep_handlers.go

```diff
@@ -50,8 +51,26 @@ func (r *ImageBasedUpgradeReconciler) launchGetSeedImage(ibu *lcav1alpha1.ImageB
 		return
 	}
 	r.Log.Info("Handler script written")
-	go r.Executor.Execute(scriptname, "--seed-image", ibu.Spec.SeedImageRef.Image, "--progress-file", progressfile, "--image-list-file", imageListFile)
 
+	// Use cluster wide pull-secret by default
+	pullSecretFilename := "/var/lib/kubelet/config.json"
+
+	if ibu.Spec.SeedImageRef.PullSecretRef != nil {
+		var pullSecret string
+		pullSecret, err = utils.LoadSecretData(ctx, r.Client, ibu.Spec.SeedImageRef.PullSecretRef.Name, ibu.Namespace, corev1.DockerConfigJsonKey)
+		if err != nil {
+			r.Log.Error(err, "Failed to retrieve seed image pull-secret")
```

Comment:
maybe wrap the error and let caller to print the log? I think logging in one place is better 
err = fmt.Errorf("failed to write seed image pull-secret to file %s, err: %w", pullSecretFilename, err)